File: Day02/part1.py
file_path = "input.txt"

sumInvalid = 0

def invalid(ID):
	length = len(str(ID))
	if length % 2 != 0:
		return False
	return ID % (10 ** (length // 2) + 1) == 0

try:
	with open(file_path, 'r') as file:
		for line in file:
			line = line.strip()
			ranges = line.split(',')

			for idRange in ranges:
				start, end = idRange.split('-')
				start = int(start)
				end = int(end)
				for i in range(start, end+1):
					if invalid(i):
						sumInvalid += i

	print(f"The answer is: {sumInvalid}")
except Exception as e:
	print(f"An error occurred: {e}")

File: Day05/part1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#define ll long long int



int main(){
	std::ifstream fin("input.txt");
	std::string line;

	if (!fin.is_open()) {
		std::cout << "Error: Cannot open file!" << std::endl;
		return 1;
	}

	std::vector<std::pair<ll, ll>> ranges;
	bool getRanges = true;
	int count = 0;

	while (std::getline(fin, line)) {
		if (line.length() == 0) {
			getRanges = false;
			continue;
		}

		if (getRanges) {
			size_t hyphenPos = line.find('-');
			ranges.push_back(std::make_pair(
				std::stoll(line.substr(0, hyphenPos)),
				std::stoll(line.substr(hyphenPos+1))
			));
		} else {
			ll id = std::stoll(line);
			for (const auto& [l, r] : ranges) {
				if (l <= id && id <= r) {
					count++;
					break;
				}
			}
		}
	}



	fin.close();

	std::cout << "The answer is: " << count << std::endl;

	return 0;
}
